source/adapters/opencl/memory.cpp

```diff
@@ -288,10 +306,18 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemImageCreate(
   cl_image_desc ImageDesc = mapURImageDescToCL(pImageDesc);
   cl_map_flags MapFlags = convertURMemFlagsToCL(flags);
 
-  *phMem = reinterpret_cast<ur_mem_handle_t>(clCreateImage(
-      cl_adapter::cast<cl_context>(hContext), MapFlags, &ImageFormat,
-      &ImageDesc, pHost, cl_adapter::cast<cl_int *>(&RetErr)));
-  CL_RETURN_ON_FAILURE(RetErr);
+  try {
+    cl_mem Mem =
+        clCreateImage(hContext->get(), MapFlags, &ImageFormat, &ImageDesc,
+                      pHost, cl_adapter::cast<cl_int *>(&RetErr));
+    CL_RETURN_ON_FAILURE(RetErr);
+    auto URMem = std::make_unique<ur_mem_handle_t_>(Mem, hContext);
+    *phMem = URMem.release();
```

Comment:
I there an advantage to using a unique_ptr here and then call `release()` on it right away? Wouldn't it be simpler and faster to just use `new`?

I know that `new` is frowned upon. But this is a case where I see little advantage in using smart pointers given that the underlying api requires raw pointers. Please correct me if I'm missing something.

Reply:
I don't understand why make_unique is safer in this scenario. The `new` operator can also throw `std::bad_alloc` and will delete the object if the constructor throws.

My understanding is that the only situation, where make_unique() would be better for exception safety is if we are allocating multiple objects in the same statement which is not the case. 

But it's not a big deal. Personally, I would prefer to keep it simple and fast by just using `new`. But I understand that this is a style preference and since there is other adapters using this style I'm ok with keeping it as is.